apps/linear_algebra/benchmarks/CMakeLists.txt

```diff
@@ -35,18 +35,16 @@ target_link_libraries(halide_benchmarks PRIVATE halide_blas)
 list(APPEND BLAS_NAMES halide)
 
 find_package(Eigen3 QUIET)
-if (NOT Eigen3_FOUND)
+if (NOT EIGEN3_FOUND)
   message(STATUS "linear_algebra: Eigen3 Missing, skipping Eigen3 benchmarks")
 else()
   add_executable(eigen_benchmarks
     eigen_benchmarks.cpp
   )
   target_compile_definitions(eigen_benchmarks PRIVATE -DEIGEN_DONT_PARALLELIZE ${MISC_DEFINITIONS})
   target_compile_options(eigen_benchmarks PRIVATE -Wno-error=unused-variable)
-  target_link_libraries(eigen_benchmarks
-    PRIVATE
-      Eigen3::Eigen
-  )
+  target_include_directories(eigen_benchmarks PRIVATE ${EIGEN3_INCLUDE_DIR})
```

Comment:
This shouldn't be necessary and should be handled by `target_link_libraries` below... see https://eigen.tuxfamily.org/dox/TopicCMakeGuide.html

Also see [here](https://bitbucket.org/eigen/eigen/src/94875feeeeb9abe5509b314197da1991ba2070f5/cmake/FindEigen3.cmake#lines-103:107) where `${EIGEN3_INCLUDE_DIR}` is added to the INTERFACE_INCLUDE_DIRECTORIES property of `Eigen3::Eigen`

Reply:
This just got a lot more interesting.  Without the target_include_directories() it (meaning your code) doesn't work for me, and with target_link_libraries(... ${Eigen3}) it fails when it tries to add "-lEigen3" to the linker, which of course doesn't exist.

I'm curious about your platform where this Eigen3::* stuff actually exists, and where Eigen is a dynamic library.  Since I found a second bug in Eigen (a missing const triggering Halide's -Werror) so I guess that will be my next stop :-(

Reply:
Ha, the -lEigen3 was my mistake.  I might have a working version now...